test/conformance/prerun.sh

```diff
@@ -0,0 +1,27 @@
+# prerun.sh sets up the test function to use the functions framework commit
+# specified. This makes the function `pack` buildable with GCF buildpacks.
+#
+# `pack` build example command:
+# pack build myfn --verbose --builder us.gcr.io/fn-img/buildpacks/ruby30/builder:ruby30_20220620_3_0_4_RC00 --env GOOGLE_RUNTIME=ruby30 --env GOOGLE_FUNCTION_TARGET=http_func --env X_GOOGLE_TARGET_PLATFORM=gcf
+FRAMEWORK_VERSION=$1
+
+# exit when any command fails
+set -e
+
+cd $(dirname $0)
+
+if [ -z "${FRAMEWORK_VERSION}" ]
+    then
+        echo "Functions Framework version required as first parameter"
+        exit 1
+fi
+
+echo "source 'https://rubygems.org'
+gem 'functions_framework', github: 'GoogleCloudPlatform/functions-framework-ruby', ref: '$FRAMEWORK_VERSION'" > Gemfile
+cat Gemfile
+
+sudo gem install bundler
```

Comment:
I don't understand how this would work. This is run as part of the buildpack-integration-test action, and the prerun script is just run directly as a step, on ubuntu-latest ([link](https://github.com/GoogleCloudPlatform/functions-framework-conformance/blob/master/.github/workflows/buildpack-integration-test.yml#L148)). I don't actually see a Ruby installation there.

Reply:
The ruby installation is handled by the GCF builder, which is what happens in GCF too. You can see a sample run here: https://github.com/anniefu/functions-framework-ruby/runs/7099652594?check_suite_focus=true

Reply:
The FF conformance test client has a buildpack mode which can orchestrate using `pack build` to build and do everything in containers instead of the local mode that the FFs use for conformance tests.

Reply:
ah okay, the indirection inherent in buildpacks still confuses me. This looks good then.